app/controllers/users_controller.rb

```diff
@@ -1,3 +1,30 @@
 class UsersController < ApplicationController
-    def new; end
+    def show
+      @user = User.find_by id: params[:id]
+      return if @user
+  
+      flash[:danger] = t ".fail_message"
+      redirect_to root_path
+    end
+
+    def new
+      @user = User.new
+    end
+
+    def create
+      @user = User.new user_params
+      if @user.save
+        flash[:success] = t ".welcome_message"
+        redirect_to @user
+      else
+        flash[:danger] = t ".fail_message"
+        render :new
+      end
+    end
+
+    private
+
+    def user_params
+      params.require(:user).permit User::USERS_PARAMS
```

Comment:
params thuộc kiểu dữ liệu gì?
require có tác dụng gì? In params.require(:user) ra đây em. 
permit có tác dụng gì? Tìm hiểu về strong parameter
Thử thêm 1 cặp key -value vào params.require(:user) rồi dùng permit xem có chuyện gì xảy ra

Reply:
params: là một đối tượng của class ActionController::Parameters thuộc kiểu dl Hash.
require: trả về giá trị value của trong cặp key-value tương ứng key truyền vào.
![image](https://user-images.githubusercontent.com/68638897/89278207-02cea600-d670-11ea-98d2-4ce977ef9bb8.png)
permit: lọc ra những thuộc tính được cho phép update.